src/load.ts

```diff
@@ -18,16 +17,19 @@ export const clearMemoizedConfig = (): void => {
 
 export const load = (
   passedConfig: MemoizedConfig = memoizedConfig
-): Record<string, any> => {
-  if (passedConfig !== undefined) {
-    memoizedConfig = passedConfig
-    return passedConfig
-  }
-
+): HydratedConfig => {
   if (R.isNil(process.env.RUNTIME_ENVIRONMENT)) {
     throw new Error('process.env.RUNTIME_ENVIRONMENT must be defined.')
   }
 
+  if (passedConfig !== undefined) {
```

Comment:
Could use the short form here:

```suggestion
  if (passedConfig) {
```

Reply:
or, for consistency with the other nil checks:

```suggestion
  if (!R.isNil(passedConfig)) {
```

Reply:
addressed